File: examples/integration/list-robots-and-tasks.py
#!/usr/bin/env python3
"""
List robots and tasks using the SAI AUROSY API.
Usage: python list-robots-and-tasks.py [base_url] [api_key]
Example: python list-robots-and-tasks.py http://localhost:8080/api/v1 sk-integration-abc123
"""
import json
import sys
import urllib.request

BASE_URL = sys.argv[1] if len(sys.argv) > 1 else "http://localhost:8080/api/v1"
API_KEY = sys.argv[2] if len(sys.argv) > 2 else ""


def request(path: str) -> dict:
    req = urllib.request.Request(
        f"{BASE_URL}{path}",
        headers={"X-API-Key": API_KEY},
    )
    with urllib.request.urlopen(req) as resp:
        return json.loads(resp.read().decode())


def main():
    if not API_KEY:
        print("Usage: python list-robots-and-tasks.py [base_url] [api_key]")
        sys.exit(1)

    print("=== Listing robots ===")
    robots = request("/robots")
    print(json.dumps(robots, indent=2))

    print("\n=== Listing tasks ===")
    tasks = request("/tasks")
    print(json.dumps(tasks, indent=2))


if __name__ == "__main__":
    main()

File: examples/integration/list-robots-and-tasks.go
// List robots and tasks using the SAI AUROSY API.
// Usage: go run list-robots-and-tasks.go [base_url] [api_key]
// Example: go run list-robots-and-tasks.go http://localhost:8080/api/v1 sk-integration-abc123
package main

import (
	"encoding/json"
	"fmt"
	"net/http"
	"os"
)

func main() {
	baseURL := "http://localhost:8080/api/v1"
	apiKey := ""
	if len(os.Args) >= 2 {
		baseURL = os.Args[1]
	}
	if len(os.Args) >= 3 {
		apiKey = os.Args[2]
	}
	if apiKey == "" {
		fmt.Println("Usage: go run list-robots-and-tasks.go [base_url] [api_key]")
		os.Exit(1)
	}

	client := &http.Client{}
	req, _ := http.NewRequest("GET", baseURL+"/robots", nil)
	req.Header.Set("X-API-Key", apiKey)

	resp, err := client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "robots: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()

	var robots []any
	if err := json.NewDecoder(resp.Body).Decode(&robots); err != nil {
		fmt.Fprintf(os.Stderr, "decode robots: %v\n", err)
		os.Exit(1)
	}
	fmt.Println("=== Listing robots ===")
	enc := json.NewEncoder(os.Stdout)
	enc.SetIndent("", "  ")
	enc.Encode(robots)

	req, _ = http.NewRequest("GET", baseURL+"/tasks", nil)
	req.Header.Set("X-API-Key", apiKey)
	resp, err = client.Do(req)
	if err != nil {
		fmt.Fprintf(os.Stderr, "tasks: %v\n", err)
		os.Exit(1)
	}
	defer resp.Body.Close()

	var tasks []any
	if err := json.NewDecoder(resp.Body).Decode(&tasks); err != nil {
		fmt.Fprintf(os.Stderr, "decode tasks: %v\n", err)
		os.Exit(1)
	}
	fmt.Println("\n=== Listing tasks ===")
	enc.Encode(tasks)
}
